thunderfish/chirp.py

```diff
@@ -0,0 +1,196 @@
+                                          noverlap=noverlap, pad_to=pad_to, sides=sides, scale_by_freq=scale_by_freq)
+    return spectrum, freqs, time
+
+def clean_chirps(chirp_time_idx, power, power_window=100):
+    """
+    Chirp is only accepted as such if the power of the frequency drops down as expected.
+
+    :param chirp_time_idx: (array) indices of chirps.
+    :param power: (array) power array containing for each timestamp the max value in power of a certain frequency range.
+    :param power_window (int) datapoints arroung a detected chirp used to verify that there is a chirp.
+    :return: chirp_time_idx: (array) indices of chirps that have been confirmed to be chirps.
+    """
+    true_chirps = np.array([], dtype=bool)
+    for i in range(len(chirp_time_idx)):
+        idx0 = int(chirp_time_idx[i] - power_window/2)
+        idx1 = int(chirp_time_idx[i] + power_window/2)
+
+        tmp_median = np.median(power[idx0:idx1])
+        tmp_std = np.std(power[idx0:idx1], ddof=1)
+
+        if np.min(power[idx0:idx1]) < tmp_median - 3*tmp_std:
+            true_chirps = np.append(true_chirps, True)
+        else:
+            true_chirps = np.append(true_chirps, False)
+    return chirp_time_idx[true_chirps]
+
+def chirp_detection(spectrum, freqs, time, fishlist, min_power= 0.005, freq_tolerance=5., chirp_th=1., plot_data=False):
+    """
+    Detects chirps on the basis of a spectrogram.
+
+    :param spectrum: (2d-array) spectrum calulated with the numpy.spectrogram function.
+    :param freqs: (array) frequencies of the spectrum.
+    :param time: (array) time of the nffts used in the spectrum.
+    :param fishlist: (array) power und frequncy for each fundamental/harmonic of a detected fish.
+                     fishlist[fish][harmonic][frequency, power]
+    :param min_power: (float) minimum power of the fundamental frequency for each fish to participate in chirp detection.
+    :param freq_tolerance: (float) frequency tolerance in the spectrum to detect the power of a certain frequency.
+    :param chirp_th: (float) minimum chirp duration to be accepted as a chirp.
+    :param plot_data: (bool) If True: plots the process of chirp detection.
+    :return:chirp_time: (array) array of times (in sec) where chirps have been detected.
+    """
+    fundamentals = []
+    for fish in fishlist:
+        if fish[0][1] > min_power:
+            fundamentals.append(fish[0][0])
+
+    chirp_time = np.array([])
+
+    for enu, fundamental in enumerate(fundamentals):
+        # extract power of only the part of the spectrum that has to be analysied for each fundamental
+        spectrum1 = spectrum[freqs >= fundamental - freq_tolerance]
+        freqs1 = freqs[freqs >= fundamental - freq_tolerance]
+        spectrum2 = spectrum1[freqs1 <= fundamental + freq_tolerance]
+        # freqs2 = freqs1[freqs1 <= fundamental + freq_tolerance]
+
+        # get the peak power of every piont in time
+        power = np.max(spectrum2[:], axis=0)
```

Comment:
Even simpler, everything upt line 113 is just:
power = np.max(spectrum[(freqs >= fundamental - freq_tolerance) & (freqs <= fundamental + freq_tolerance)], axis=0)